d2l-table-wrapper.js

```diff
@@ -0,0 +1,317 @@
+/**
+# D2L Table
+
+## Usage
+```
+	<link rel="import" href="d2l-table.html">
+	<d2l-table-wrapper>
+		<table class="d2l-table">
+		<thead>
+			<tr>
+				<th>Table header cell</th>
+			</tr>
+		</thead>
+		<tfoot>
+			<tr>
+				<th>Table header cell</th>
+			</tr>
+		</tfoot>
+		<tbody>
+			<tr>
+				<td>Table cell</td>
+			</tr>
+		</tbody>
+		</table>
+	</d2l-table-wrapper>
+```
+
+### Table attributes
+
+Attribute | Description
+----------|-------------
+`selectable` | Add hover effect for rows
+
+### Row Attributes
+
+Attribute | Description
+----------|-------------
+`selected` | Apply selected style
+`active` | Apply active style
+`header` | Apply header style
+`last-row` | This row is to be treated as the bottom row in the case where CSS can't detect it. This should only happen if the last row(s) are hidden from view, and the table has no footer
+
+## Styling
+
+Custom property | Description | Default
+----------------|-------------|----------
+`--d2l-table-border-color` | Border Color | `var(--d2l-color-mica)` |
+`--d2l-table-border-radius` | Corner Border Radius | `0.3rem` |
+`--d2l-table-border` | Border | `1px solid var(--d2l-table-border-color);` |
+`--d2l-table-header-background-color` | Header background color (th elements under `<thead>` or `<tr header>`) | `var(--d2l-color-regolith);` |
+`--d2l-table-body-background-color` | Body background color (non-header) | `#fff` |
+`--d2l-table-row-background-color-active` | Active row background color | `var(--d2l-color-celestine-plus-2)` |
+`--d2l-table-row-border-color-selected` | Selected row border color | `var(--d2l-color-celestine-plus-1)` |
+`--d2l-table-row-border-color-active-selected` | Active and Selected row border color | `var(--d2l-color-celestine-plus-1)` |
+`--d2l-table-row-background-color-active-selected` | Active and Selected row background color | `#EBF5FC` |
+`--d2l-table-row-background-color-selected` | Selected row background color | `var(--d2l-color-celestine-plus-2)` |
+`--d2l-table-border-overflow` | Border to show when the table overflows | `dashed 1px #d3d9e3` |
+
+```
+	<link rel="import" href="d2l-table.html">
```

Comment:
The readme and documentation will need to be updated. Should this be done before merging or after? So this PR can focus on functionality?

Reply:
Yeah, the plan was to worry about fixing up the READMEs after the merge.  I exclude README changes in the script so if we do make updates to it and then re-run the script, the README won't get overwritten.  I can't really do this the same way in the actual script documentation, but I'm wondering if all of this should just live in the README anyways